backend/tasks/scheduler.go

```diff
@@ -52,32 +53,36 @@ func (jr *JobRunner) generateOpenContentProviderTasks() ([]models.RunnableTask,
 		log.Errorf("failed to fetch all open content providers: %v", err)
 		return nil, err
 	}
-	for idx := range providers {
-		for _, jobType := range models.AllContentProviderJobs {
-			switch jobType {
-			case models.ScrapeKiwixJob:
-				job := models.CronJob{Name: string(jobType)}
-				if providers[idx].Title == models.Kiwix {
-					task, err := jr.handleCreateOCProviderTask(&job, providers[idx].ID)
-					if err != nil {
-						log.Errorf("failed to create task: %v", err)
-						continue
-					}
-					otherTasks = append(otherTasks, *task)
-				}
-			case models.RetryVideoDownloadsJob, models.PutVideoMetadataJob, models.SyncVideoMetadataJob:
-				job := models.CronJob{Name: string(jobType)}
-				if providers[idx].Title == models.Youtube {
-					task, err := jr.handleCreateOCProviderTask(&job, providers[idx].ID)
-					if err != nil {
-						log.Errorf("failed to create task: %v", err)
-						continue
-					}
-					otherTasks = append(otherTasks, *task)
-				}
-			default:
+	vidProvider := slices.IndexFunc(providers, func(p models.OpenContentProvider) bool {
+		return p.Title == models.Youtube
+	})
+	libProvider := slices.IndexFunc(providers, func(p models.OpenContentProvider) bool {
+		return p.Title == models.Kiwix
+	})
+	for _, jobType := range models.AllContentProviderJobs {
+		if jobType.IsLibraryJob() {
+			if libProvider == -1 {
 				continue
 			}
+			job := models.CronJob{Name: string(jobType)}
+			task, err := jr.handleCreateOCProviderTask(&job, providers[libProvider].ID)
+			if err != nil {
+				log.Errorf("failed to create task: %v", err)
+				continue
+			}
+			otherTasks = append(otherTasks, *task)
+		}
+		if jobType.IsVideoJob() {
```

Comment:
Should this be pulled up and made into an else if statement? or be changed to a switch block? This if check will always be checked if the first condition is true.

Reply:
yeah true either that or a continue after appending, but i like the else if better :+1: